components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java

```
@@ -259,24 +285,33 @@ public CertificateMetadataDTO getCertificate(String alias, String endpoint, int
-            connection.commit();
-            preparedStatement = connection.prepareStatement(getCertQuery);
-            preparedStatement.setInt(1, tenantId);
-            preparedStatement.setString(2, alias);
-            preparedStatement.setString(3, endpoint);
-            resultSet = preparedStatement.executeQuery();
+            try {
+                connection = APIMgtDBUtil.getConnection();
+                initialAutoCommit = connection.getAutoCommit();
+                connection.setAutoCommit(false);
+                connection.commit();
+                preparedStatement = connection.prepareStatement(getCertQuery);
+                preparedStatement.setInt(1, tenantId);
+                preparedStatement.setString(2, alias);
+                preparedStatement.setString(3, endpoint);
+                resultSet = preparedStatement.executeQuery();
 
-            while (resultSet.next()) {
-                certificateMetadataDTO = new CertificateMetadataDTO();
-                certificateMetadataDTO.setAlias(resultSet.getString("ALIAS"));
-                certificateMetadataDTO.setEndpoint(resultSet.getString("END_POINT"));
+                while (resultSet.next()) {
+                    certificateMetadataDTO = new CertificateMetadataDTO();
+                    certificateMetadataDTO.setAlias(resultSet.getString("ALIAS"));
+                    certificateMetadataDTO.setEndpoint(resultSet.getString("END_POINT"));
+                }
+            } catch (SQLException e) {
+                handleException("Error while retrieving certificate metadata.", e);
+            } finally {
+                APIMgtDBUtil.closeStatement(preparedStatement);
+                APIMgtDBUtil.closeAllConnections(preparedStatement, connection, resultSet);
+                connection.setAutoCommit(initialAutoCommit);
             }
         } catch (SQLException e) {
-            handleException("Error while retrieving certificate metadata.", e);
```

Exception swallowed?

---

fixed with https://github.com/wso2/carbon-apimgt/commit/3c8d1e61dc642ebe8179c483156798312a03382b